backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/ExternalSystemConfigRepo.java

```diff
@@ -863,46 +866,60 @@ private ExternalSystemLeichMehlConfig buildExternalSystemLeichMehlConfig(@NonNul
 				.tcpPort(config.getTCP_PortNumber())
 				.tcpHost(config.getTCP_Host())
 				.pluFileExportAuditEnabled(config.isPluFileExportAuditEnabled())
-				.productMappings(getExternalSystemLeichMehlConfigProductMappings(id))
-				.pluFileConfigs(getExternalSystemLeichMehlPluFileConfigs(id))
 				.build();
 	}
 
 	@NonNull
-	private List<ExternalSystemLeichMehlConfigProductMapping> getExternalSystemLeichMehlConfigProductMappings(@NonNull final ExternalSystemLeichMehlConfigId externalSystemLeichMehlConfigId)
+	public Optional<ExternalSystemLeichMehlConfigProductMapping> getExternalSystemLeichMehlConfigProductMappings(@NonNull final ProductId productId, @Nullable final BPartnerId bPartnerId)
```

Comment:
returning more than one root entity from a Repository is a sign that the repository shall be split in 2 Repositories.

i.e pls introduce ExternalSystemLeichMehlConfigProductMappingRepository